[DPS-2773] Improvements to the Building webhook apps article (#869)
livechat/livechat-public-docs: content/getting-started/livechat-apps/webhook-apps/index.mdx

1. Imports [Sign in with LiveChat flow](/getting-started/authorization/sign-in-with-livechat/).
2. Imports your sign in form or other authorization flow (e.g. "Connect with..." button).
3. Binds both pieces of information and stores the new user in your database.
4. Displays a confirmation screen when the integration is up and running.

The settings page will appear right after LiveChat user installs your app from the Marketplace.

## 6. Register the webhooks

Once you have the LiveChat credentials, you can register your webhooks with the Configuration API. Currently, there are two APIs you can use to manage webhooks:

- [Configuration API v3](/management/configuration-api/#webhooks)
- [Configuration API v2](/management/configuration-api/v2.0/#create-a-new-webhook) (deprecated)

### Seting up webhooks for development

One of the possible development options is to set up your webhooks from within the LiveChat Agent App. However, please note that this configuration uses [Configuration API v2 webhooks](/management/configuration-api/v2.0/#create-a-new-webhook), which are now deprecated.

> We highly recommend using the [Configuration API v3 webhooks](/management/configuration-api/#webhooks). To view a tutorial on setting these up, follow [this link](/getting-started/app-guides/chat-webhooks/#tutorial).

Reviewer: ```suggestion
We highly recommend using the [Configuration API v3 webhooks](/management/configuration-api/#webhooks).

Reviewer: It's better not to link that at the moment. v3.3 introduces important changes that we'll need to include in this docs

To configure webhooks v2 within the LiveChat Agent App, go to **Settings > Integrations > [Webhooks](https://my.livechatinc.com/settings/webhooks)** and set up your webhook URLs.

![LiveChat Webhooks](./images/livechat-platform-webhooks-settings.png)

Each webhook consists of the following properties:

- **Event** – determines when the webhook is sent to your web server.
- **Data type** – allows you to select which information will be included in the webhook.

> - **Target URL** – address of your web server the webhook will be sent to.

Reviewer: ```suggestion
- **Target URL** – the address of your web server the webhook will be sent to.
```

# Webhooks reference

Please see the following documents for reference on how to manage webhooks:

- How to [set up](/management/configuration-api/#register-webhook) webhooks
- A list of available webhooks along with their [format samples](/management/configuration-api/#webhooks)

## Webhooks v2 reference (deprecated)

<Warning>

**Note: This reference is deprecated**. Versions 3.x of Platform APIs introduce new webhook interfaces. We highly recommend migrating out of the deprecated system. The new webhooks service has much more functionality and provides you with a higher control over the data. Please note that the new Platform APIs also provide a **RTM (websocket) transport**.

</Warning>

> LiveChat can send notifications when some specific action is performed. Such a notification is called a **webhook** – it’s a simple HTTP request that LiveChat sends to your server when a particular event occurs. You can find a basic auth example below.

Reviewer: ```suggestion
LiveChat can send notifications when some specific actions are performed. Such a notification is called a **webhook** – it’s a simple HTTP request that LiveChat sends to your server when a particular event occurs. You can find a basic auth example below.
```

## 2. Create a new app

Once you register an account, create a new application in the Developer Console using the `Create new app` button.

![LiveChat Developer Console](./images/livechat-developers-console-create-app.png)

## 3. Add Agent App Widgets block

Go to the Building Blocks section and create a new Agent App Widget with `App Settings` as the placement. This widget will be displayed as a settings page for your app.

![Create new Settings widget](./images/livechat-developers-console-create-widget.png)

## 4. Add the App Authorization block

Go back to the Building Blocks section and add the `Server-side app` option. Save the highlighted `Client Id` &ndash; you'll need it later to complete the authorization.

![Create new server-side Authorization block](./images/livechat-developers-console-create-auth.png)

## 5. Create a settings page

Settings page is a place where you can:

- Prompt users to connect their accounts with your OAuth2.1 provider.
- Display configuration options.
- Inform users about additional installation steps (if there are any).

> If you need to access LiveChat user data, we recommend using [Sign in with LiveChat authorization flow](/getting-started/authorization/sign-in-with-livechat/). For more information on possible widget placements, visit the [app locations](/extending-agent-app/#app-locations) page.

Reviewer: ```suggestion
If you need to access LiveChat user data, we recommend using the [Sign in with LiveChat flow](/getting-started/authorization/sign-in-with-livechat/). For more information on possible widget placements, visit the [app locations](/extending-agent-app/#app-locations) page.
```

### Sample settings page scenario

If you're building an integration that binds LiveChat data with an external service, you should create a settings page that:

1. Makes use of the [Sign in with LiveChat flow](/getting-started/authorization/sign-in-with-livechat/).

> 2. Imports your sign in form or other authorization flow (e.g. "Connect with..." button).

Reviewer: + link to https://developers.livechat.com/docs/authorization/authorizing-api-calls/#authorization-code-grant

Each webhook contains the following properties:

- `event_type` – tells you the event that triggered the webhook. Possible values: `chat_started`, `chat_ended`, `visitor_queued`.

> - `token` and `license_id` – your authentication credentials that let you call LiveChat’s Configuration API v2 methods. You won’t need to use them unless you want to make a call to LiveChat’s API right when you receive a webhook. In that case, you just need to pass the `token` and `license_id` credentials in your API call.

Reviewer: ```suggestion
- `token` and `license_id` – your authentication credentials that let you call LiveChat Configuration API v2 methods. You won’t need to use them unless you want to make a call to the LiveChat API right when you receive a webhook. In that case, you just need to pass the `token` and `license_id` credentials in your API call.
```

Note: LiveChat webhooks are sent with `Content-Type: application/json` header, so please make sure that your service can handle such requests.

> Note: `chat_changed` event type is sent only when the tag list for a particular chat was changed, no matter if the chat is currently ongoing or not.

Reviewer: I wonder what the author originally meant by "pending"? 🤔 Queued, maybe?

Author: As far as I remember the chat_changed webhook triggers anytime a tag is changed, meaning during or after the chat - so the ongoing (in my mind) meant that the chat is currently live, not already archived :D

</CodeResponse>

In some cases, you may want to get some additional information when a particular event occurs.

> For example, when the chat starts, you may want to know the exact chat start time along with the visitor's name and e-mail provided in the pre-chat survey. To get this information, you can add additional **data types** that will be sent to your web server along with each webhook.

Reviewer: ```suggestion
For example, when the chat starts, you may want to know the exact chat start time along with the visitor's name and email provided in the pre-chat survey. To get this information, you can add additional **data types** that will be sent to your web server along with each webhook.
```
